stores/metadata.go

```diff
@@ -2165,6 +2169,25 @@ func deleteObjects(tx *gorm.DB, bucket string, path string) (numDeleted int64, _
 
 func invalidateSlabHealthByFCID(tx *gorm.DB, fcids []fileContractID) error {
 	return tx.Exec(`
+	UPDATE slabs SET health_valid = 0 WHERE id in (
```

Comment:
would get rid of this function in favour of the method on the `SQLStore`? Seems like we're needlessly duplicating it, we can make `archiveContracts` a method on the store and remove this I think

Reply:
There is a bit of a difference between those two. The function is called on an existing transaction which is meant to be rolled back if it fails. So using the method will create nested transactions and block the whole thing for longer.
The method is meant to be called outside a transaction since it creates its own.